crime-assessment-service/src/main/java/uk/gov/justice/laa/crime/assessmentservice/iojappeal/validator/ApiCreateIojAppealRequestValidator.java

```diff
@@ -0,0 +1,100 @@
+    private static final String FIELD_IS_MISSING = "%s is missing.";
+
+    public List<String> validateRequest(ApiCreateIojAppealRequest request) {
+        var errorList = new ArrayList<String>();
+
+        // check the IojAppealMetadata Object.
+        errorList.addAll(validateMetaData(request));
+        // check the IojAppeal Object.
+        errorList.addAll(validateIoJAppeal(request));
+
+        return errorList;
+    }
+
+    // MetaData Validation
+
+    List<String> validateMetaData(ApiCreateIojAppealRequest request) {
+        List<String> errorList = new ArrayList<>();
+        IojAppealMetadata metadata = request.getIojAppealMetadata();
+
+        if (Objects.isNull(metadata)) {
+            errorList.add(String.format(FIELD_IS_MISSING, "Metadata"));
+            return errorList;
+        }
+        // nullchecks
+        validateFieldNotNull(metadata.getCaseManagementUnitId(), "Case Management Unit", errorList);
+        validateFieldNotNull(metadata.getUserSession(), "User Session", errorList);
+        validateAnApplicationIdPresent(metadata, errorList);
+        return errorList;
+    }
+
+    // Ensure that we have an application ID. Either LegacyApplicationId, or ApplicationId must be present.
+    private void validateAnApplicationIdPresent(IojAppealMetadata metadata, List<String> errorList) {
+        if (Objects.isNull(metadata.getApplicationId()) && Objects.isNull(metadata.getLegacyApplicationId())) {
+            errorList.add(String.format(FIELD_IS_MISSING, "Both Application Id and Legacy Application Id"));
+        }
+    }
+
+    // Appeal Validation
+
+    List<String> validateIoJAppeal(ApiCreateIojAppealRequest request) {
+        List<String> errorList = new ArrayList<>();
+
+        IojAppeal appeal = request.getIojAppeal();
+        if (Objects.isNull(appeal)) {
+            errorList.add(String.format(FIELD_IS_MISSING, "Appeal"));
+            return errorList;
+        }
+        // validate notNull
+        validateFieldNotNull(appeal.getAppealDecision(), "Appeal Decision", errorList);
+        validateFieldNotNull(appeal.getAppealAssessor(), "Appeal Assessor", errorList);
+        validateFieldNotNull(appeal.getAppealReason(), "Appeal Reason", errorList);
+        validateFieldNotNull(appeal.getDecisionReason(), "Decision Reason", errorList);
+        validateFieldNotNull(appeal.getDecisionDate(), "Decision Date", errorList);
+        validateFieldNotNull(appeal.getReceivedDate(), "Received Date", errorList);
+        // validate content
+        validateAppealReasonType(appeal.getAppealReason(), errorList);
+        validateAppealReasonAppealAssessorCombinations(appeal, errorList);
+        return errorList;
+    }
+
+    // check the appeal reason is of type HARDIOJ
+    private void validateAppealReasonType(NewWorkReason appealReason, List<String> errorList) {
+        if (Objects.nonNull(appealReason) && !appealReason.getType().equalsIgnoreCase("HARDIOJ")) {
```

Comment:
Do we need to check nonNull here for appealReason as it has already been checked on L66. Feels like we don't need to do it twice.